research-software-directory/RSD-as-a-service
# 770 extra git data

# Scrape more Git data

Changes proposed in this pull request:

* Add a scraper that scrapes for GitHub and GitLab the star count, fork count and open issue count (GitHub only), the license scraper is merged into this scraper
* Add a scraper that scrapes for GitHub and GitLab the contributor count
* Some refactoring and renaming of the scrapers

How to test:
* `docker-compose down --volumes && docker-compose build --parallel && docker-compose up --scale data-generation=1`.
* Wait for all the scrapers to run or run all Git scrapers with the following commands:
```bash
docker-compose exec scrapers java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.git.MainProgrammingLanguages
docker-compose exec scrapers java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.git.MainContributors
docker-compose exec scrapers java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.git.MainBasicData
docker-compose exec scrapers java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.git.MainCommits
```
Use a tool like DBeaver to check that data for all categories have been scraped (`SELECT * FROM repository_url;`)

Closes #770

PR Checklist:
